content/francineblanc/how-to-use-git-for-beginners.md

```diff
@@ -0,0 +1,278 @@
+
+![Image of git status command being run after file in staging area](git-add-and-status.png)
+
+### Unstaging a file
+
+There may be occasions where a file has been incorrectly added to the staging area. This can be undone with the command `git reset <filename>`. This will not affect the changes done to the file in any way, it will simply remove it from the staging area. 
+
+In the case where all files need to be unstaged, the command [`git reset`] can be run without adding further arguments.
+
+As of version 2.24, there is an alternative syntax for unstaging files in Git: 
+
+```pseudo
+git restore --staged <filename>
+``` 
+
+The `<filename>` is the name of the file to be unstaged.
+
+## Committing changes
+
+Committing is generally the last step of the Git workflow and is generally thought of as being a snapshot of a project at a particular time. In this stage, changes in the staging area are saved inside the local repo, which is achieved with the command `git commit -m` followed by a space and a short message explaining the commit in quotes. For example:
+
+```bash
+git commit -m "Add title and description to intro file"
+```
+
+There are general [conventions] for writing a commit message including: 
+- Making sure they are no more than 50 characters long.
+- Writing them in the imperative or present tense. 
+
+Messages should always be clear and informative to others who may go back to confirm what changes were made.
+
+Committing is an important step in the Git workflow, as once a change has been committed, it can be recalled at a later date or the project can be rewound to that particular version. Commit history can be viewed with `git log`, which will display a list of commits in chronological order (the most recent commits being at the top), along with information such as the author of the commit, the date and time of the commit and the commit message.
+
+### Undoing a commit
+
+In the situation where the last commit needs to be undone, the `git reset` command can again be used, with some modifications. 
+
+The current commit is called the `HEAD` commit, which is generally the most recently made commit. To find out which commit this is, the command `git show HEAD` can be executed in the terminal. This will display information about the most recent commit, including a unique 40 character SHA hash, which Git uses like an id to identify revisions in the repo.
+
+To undo the immediately previous commit, the command `git reset --soft HEAD~1` can be run, which will rewind the current `HEAD` commit to the immediately previous commit. The `--soft` flag ensures that any changes made to the files are preserved.
+
+If the rewind needs to go beyond the most recent commit, the command `git reset SHA`, where `SHA` is the first 7 characters of the SHA of the commit, can be used. Note that the SHA of all previous commits can be viewed using `git log`.
```

Comment:
For these lines, I would place a \```bash snippet between them where it contains the `git reset` command:

To undo the direct previous commit, the following command can be run:

```bash
git reset --soft HEAD~1
```

This will repoint the current `HEAD` commit to the one immediately before it. If we need to go beyond the most recent commit, we can run `git reset SHA`, where `SHA` is the first 7 characters of the commit hash.

Reply:
> Note that the SHA of all previous commits can be viewed using `git log`.

This info should be incorporated somewhere in line # 157.